src/charm.py

```diff
@@ -303,6 +302,19 @@ def logs_rules(self) -> Dict[str, Any]:
         """Return a list of logging rules."""
         return self._cos.logs_alerts
 
+    @property
+    def requested_receivers(self) -> set:
+        """Return a list of requested tracing receivers."""
+        protocols = self._cos.requested_protocols()
+        protocols.update(
+            [
+                receiver
+                for receiver in get_args(ReceiverProtocol)
+                if self.config.get(f"always_enable_{receiver}") is True
+            ]
+        )
+        return protocols
```

Comment:
last but not least, reading this method made me realize we have a maintenance-wise dangerous disconnect between the config options here and the values in `ReceiverProtocol`.
What if the tracing lib in the future adds or removes protocols from there? What do we do with the config options? Will we even notice?
Least we should do is add a test that verifies that the config.yaml has `always_enable_x` for all (and only) the protocols contained in the tracing lib we're shipping.
Then, if and when there is a risk of desyncing, we get to make a conscious decision about what to do. Otherwise we risk to forget or not even notice when that happens.

Reply:
Yeah, I think we can go one step further and compare the supported protocols (not only through overrides / config properties) in both libs as well in a test here. I'll cook something up.

Reply:
Added some tests, but they'll pass once we merge canonical/tempo-k8s-operator#149. Will get back to it tomorrow and either disable the tests for now or fetch-lib once merged.

Reply:
#149 is merged

Reply:
Update pushed and these tests passed :)